### PR TITLE
fix: validate DOI path, sort datasets by cell count

### PR DESCRIPTION
**Pull request by seve**
Functional review: @Bento007 

### Changes
* modify
  * Changes sorting for datasets to order by descending cell count
  * Add path checking for DOI link where path is not empty
  
  ---
  #1110 #1142

### CODE REVIEW

**Review by Bento007 — COMMENTED** [review]
Can you add some testing for this

File: frontend/src/components/CreateCollectionModal/components/Content/components/LinkInput/index.tsx [hunk 1]
```diff
@@ -140,14 +140,16 @@ const LinkInput: FC<Props> = ({
...
   type: COLLECTION_LINK_TYPE
 ): (value: string) => true | string {
   return (value: string) => {
     // Skip validation if type is not DOI
     if (type !== COLLECTION_LINK_TYPE.DOI) return true;
-
-    const isValid = value.includes("doi.org");
+    const origin = "doi.org/";
+    const originIndex = value.indexOf("doi.org/");
+    const isValid =
```
Comment (tihuan) [inline frontend/src/components/CreateCollectionModal/components/Content/components/LinkInput/index.tsx:151]: Maybe regex? e.g. `/doi.org\d+/` (not tested 😆)

Comment (seve) [reply frontend/src/components/CreateCollectionModal/components/Content/components/LinkInput/index.tsx:151]: I'm going to leave it as is for now to keep it simple. If we move to actually validate the DOI itself I'll move to regex

Comment (tihuan) [reply frontend/src/components/CreateCollectionModal/components/Content/components/LinkInput/index.tsx:151]: sounds good!

**Review by tihuan — APPROVED** [review]
LGTM!!

**Review by Bento007 — APPROVED** [review]

**Pull request merged**